src/drivers/common/device.c

```diff
@@ -35,6 +35,10 @@ struct dev_module *dev_module_create(struct device *dev, const char *name,
 	assert(privdata); /* No known devices without privdata. Currently
 			   *  it's up to legacy of the old FS */
 
+	while (*name == '/') {
```

Comment:
*name++ ?

Reply:
This way it will cause unwanted side-effect when `*name != '/'`.

Reply:
I don't understand why. 
```
    while (*name++ != '/')
        ;
```
works well when there are '/' in name at any place.

By the way, in any case, what if the name do not contain '/'? Or this function assumes it does anyway?

Reply:
Code you provided doesn't work well, that's exactly why another way is used :)

Just compile following code and look at the output
```
#include <stdio.h>

char *trim1(char *name) {
	while (*name == '/') { name++; }
	return name;
}

char *trim2(char *name) {
	while (*name++ == '/');
	return name;
}

int main() {
	printf("%s\n%s\n%s\n%s\n",
		trim1("/123"),
		trim2("/123"),
		trim1("123"),
		trim2("123"));
	return 0;
}
```

Reply:
@alexkalmuk it's `while (*name++ == '/')`, not `while (*name++ != '/')`

Reply:
My mistake